plugins/importexport/onix30/filter/MonographONIX30XmlFilter.php

```diff
@@ -380,20 +380,22 @@ public function createProductNode(DOMDocument $doc, Submission $submission, Publ
             $contributorNode = $doc->createElementNS($deployment->getNamespace(), 'Contributor');
             $contributorNode->appendChild($this->buildTextNode($doc, 'SequenceNumber', $sequence));
 
-            $userGroup = UserGroup::find($author->getUserGroupId());
-
-            $userGroupOnixMap = [
-                'default.groups.name.author' => 'A01',
-                'default.groups.name.volumeEditor' => 'B01',
-                'default.groups.name.chapterAuthor' => 'A01',
-                'default.groups.name.translator' => 'B06',
-                'default.groups.name.editor' => 'B21'
-            ]; // From List17, ContributorRole types.
-
-            $nameKey = $userGroup->nameLocaleKey;
-            $role = array_key_exists($nameKey, $userGroupOnixMap) ? $userGroupOnixMap[$nameKey] : 'Z99'; // Z99 - unknown contributor type.
-
-            $contributorNode->appendChild($this->buildTextNode($doc, 'ContributorRole', $role));
+            collect($author->getContributorRoleIdentifiers())
+                ->map(
+                    fn (string $contributorRoleIdentifier): string =>
+                    // From List17, ContributorRole types.
+                    match ($contributorRoleIdentifier) {
+                        ContributorRoleIdentifier::AUTHOR->getName() => 'A01',
+                        ContributorRoleIdentifier::EDITOR->getName() => 'B01',
+                        ContributorRoleIdentifier::TRANSLATOR->getName() => 'B06',
+                        default => 'Z99' /* Z99 - unknown contributor role type. */
```

Comment:
Just a note for me if I forget it later:
We decided to remove B21, because submitting user can never has press editor role.
Also we decided to use B01 for any kind of editor -- it is generic enough also if press manager would create additional EDITOR contributor roles